packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/datebox.mask.tests.js

```diff
@@ -1266,6 +1266,45 @@ module('Date AM/PM Handling', setupModule, () => {
     });
 });
 
+module('TimeZone Handling', setupModule, () => {
+    module('regex tests', () => {
+        test('should support \'x\' in date pattern and not generate errors (T1241387)', function(assert) {
+            try {
+                this.instance.option({
+                    displayFormat: 'yyyy-MM-dd\'T\'HH:mm:ssxxx',
+                    useMaskBehavior: true,
+                    type: 'date',
+                });
+                assert.ok(true, 'no error shown');
+            } catch(e) {
+                assert.ok(false, 'error exists');
+            }
+        });
+    });
+    test('should not show error when changing timezone on runtime via up/down buttons (T1241387)', function(assert) {
+        try {
+            this.instance.option({
+                displayFormat: 'yyyy-MM-dd\'T\'HH:mm:ssxxx',
+                useMaskBehavior: true,
+                type: 'date',
+                value: new Date(),
+            });
+            this.$input.trigger('dxclick');
+            this.keyboard.press('right')
+                .press('right')
+                .press('right')
+                .press('right')
+                .press('right')
+                .press('right');
```

Comment:
[refactoring]
You can use `caret` method to set a correct input selection.

Reply:
[3c3e99e](https://github.com/DevExpress/DevExtreme/pull/27770/commits/3c3e99eba324815ee381b3c900d53123cfa91ac8)